src/app/utils/isObject.ts

```diff
@@ -0,0 +1,4 @@
+const isObject = (value: any): value is Record<string, unknown> =>
+  Array.isArray(value) || value?.constructor?.name === 'Object'
```

Comment:
This type assertion is incorrect, since we are also checking if the value is an array it could also return an `Array` type. A more correct type would be:

```suggestion
const isObject = (value: any): value is Record<string, unknown> | Array<unknown> =>
  Array.isArray(value) || value?.constructor?.name === 'Object'
```

I would still highly recommended changing this back to `object` and remove the array logic entirely. The reason is that with this type the compiler cannot discriminate the primitive, which means the returned value becomes a union. See for example the `getGeometry` method:

![image](https://user-images.githubusercontent.com/695720/108061290-604ca200-7058-11eb-9bdf-209b1d329b5c.png)

Changing the implementation to purely asserting the `object` primitive allows the type to be discriminated properly:

```suggestion
const isObject = (value: any): value is object => value !== null && typeof value === 'object'
```

![image](https://user-images.githubusercontent.com/695720/108061683-f41e6e00-7058-11eb-8e54-7d5d49640e66.png)

Reply:
Reverted to the original implementation